### PR TITLE
[RFR] Button model [bug + refactor + blackify]

### LINKED ISSUE #8346: Button object instantiation mis-matching values
Button Group -
```
buttongroup = appliance.collections.button_groups.create(
    text=fauxfactory.gen_alphanumeric(),
    hover="btn_desc_{}".format(fauxfactory.gen_alphanumeric()),
    type=appliance.collections.button_groups.SERVICE)
```

Button Code:
```
button = buttongroup.buttons.create(
                 text=fauxfactory.gen_alphanumeric(),
                 hover=fauxfactory.gen_alphanumeric(),
                 dialog="6GmselZzQq",
                 system="Request",
                 request="InspectMe", )
```

Output:
```
In [13]: button
Out[13]: DefaultButton(
    group=ButtonGroup(text='DiVgyswCBf', hover='btn_desc_3rH4oGHfPL', type='Service', image='fa-user', display=None, icon_color=None), 
    text='2YcIyZY3uJ', hover=None, image='NAsaUSNH25', dialog='fa-user',
    display='6GmselZzQq', system='Request', request='InspectMe', open_url=None,
    attributes=None, visibility=None, enablement=None)

```

### PR DESCRIPTION
**Pull request by digitronik**
Purpose or Intent
=================
1. Bug having hover and image mismatch found by @sbulage https://github.com/ManageIQ/integration_tests/issues/8346
2. Refactor model removed some 5.8 requirements added arguments etc...
3. Blackify

{{pytest: cfme/tests/automate/custom_button/ -v}}

FIXES #8346

### CODE REVIEW

**Comment by digitronik** [comment]
PRT failures not related to this changes. With this complete run I found much more interesting thing related to `test_custom_button_expression` but I would like to add separate PR. I am planning one job also for overall custom button changes.

File: cfme/automate/buttons.py [hunk 1]
```diff
@@ -269,6 +299,23 @@ class DefaultButton(BaseButton):
...
+
+    Args:
+        group: Button Group object
+        text: The button name.
+        hover: The button hover text.
+        image: Icon of button.
+        playbook_cat_item: Playbook catalog item.
+        display: Button name display on Button.
+        inventory: Pointing to host
+        host: Specific host
```
Comment (mshriver) [inline cfme/automate/buttons.py]: `hosts`

also `system` is missing

Comment (mshriver) [reply cfme/automate/buttons.py]: pushed commit to fix this

**Comment by digitronik** [comment]
@mshriver  I added changes..

**Pull request merged**